### PR TITLE
[DEV APPROVED] TP-9812: Restrict whatsapp integration

### PR DESCRIPTION
**Pull request by scruti**
[TP-9812](https://moneyadviceservice.tpondemand.com/entity/9812-whatsapp-integration-restrict-to-specific-content)
Whatsapp footer on mobile devices should only appear on:
- Debt and Borrowing content.
- Pension and Retirement content.

Note: There are a big amount of new VCR cassettes recorded as I tested the category filtering with mutiple cases to have a decent pool. Open for improvements here.

### CODE REVIEW

File: app/views/shared/_contact_panels.html.erb [hunk 1]
```diff
@@ -46,43 +46,45 @@
         <p class="smallprint t-welsh-smallprint"><%= @footer.web_chat.small_print %></p>
       </div>
     </div>
-    <div class="l-contact-panel contact-panel__dynamic-content contact-panel__whatsapp">
-      <div class="contact-panel contact-panel__chat-open" id="js-whatsapp-content">
-        <%= heading_tag level: 2, class: 'contact-panel__heading contact-panel-theme-white' do %>
-        <%= render 'shared/svg/whats_app_logo' %>WhatsApp
-        <% end %>
+    <% if pensions_and_retirement_page? || debt_and_borrowing_page? %>
```
Comment (scruti) [inline app/views/shared/_contact_panels.html.erb]: Highlighting the only real code change in the file :point_up:

File: features/step_definitions/whatsapp_steps.rb [hunk 2]
```diff
@@ -1,21 +1,35 @@
+When(/^I visit the category ([\w\-]+) in (English|Welsh)$/) do |category, language|
+  locale = case language
```
Comment (MarcoBgn) [inline features/step_definitions/whatsapp_steps.rb]: Pretty subjective, but could use one line here: `locale = language == 'Welsh' ? 'cy' : 'en'`

Comment (scruti) [reply features/step_definitions/whatsapp_steps.rb]: nice! will change it!

File: features/whatsapp.feature [hunk 3]
```diff
@@ -12,23 +12,48 @@ Feature: WhatsApp Chat
...
+    When I visit the category <category> in English
+    Then I should not see the WhatsApp chat
+
+  Examples:
+    | category |
+    | homes    |
+    | cars     |
+    | family   |
+
+
```
Comment (margOnline) [inline features/whatsapp.feature]: remove extra line

**Pull request merged**